### DISCUSSION TITLE
Getting VM Working

### DISCUSSION THREAD

**Discussion by mkikets99** [krglaws/MyLFS#13] [Show and tell]
On Oracle VM VirtualBox:
![image](https://user-images.githubusercontent.com/22869811/168487947-0753fd43-42c7-4c7c-908f-abe26ac503d7.png)

**Comment by mkikets99** [comment]
Ok i founded whats wrong...
It cannot boot because the `initramfs` is missing

**Reply by mkikets99** [reply]
and yes... it is in BLFS:
https://www.linuxfromscratch.org/blfs/view/svn/postlfs/initramfs.html

**Comment by mkikets99** [comment]
Also, KVM/QEMU:
![image](https://user-images.githubusercontent.com/22869811/168489808-ef518647-c11c-436f-afc2-26c71cddedbd.png)

**Comment by mkikets99** [comment]
@krglaws Yes, it finally boots from VirtualBox
![image](https://user-images.githubusercontent.com/22869811/168593955-5cd005d9-5081-4d53-8ebd-4c167b9836eb.png)

**Reply by mkikets99** [reply]
I'll add a conversion command to it
Also.. there is an issue with grub cause it is not adding a `$PARTITIONUUID` to the `grub.cfg` file

**Reply by mkikets99** [reply]
This is bacause it must take 
```bash
LFSPARTUUID=$(lsblk -o PARTUUID $LOOP_P1 | tail -1) # needed for grub.cfg
```
and not:
```bash
LFSPARTUUID=$(lsblk -o PARTUUID $LOOP | tail -1) # needed for grub.cfg
```
Second will fail sometimes...

**Reply by krglaws** [reply]
Sorry, I missed this discussion.

It actually does not matter whether you read from $LOOP or $LOOP_P1. They will both produce the correct output. The issue was that the kernel sometimes needs a few seconds to sync itself with the newly created partitions. grub.cfg's PARTUUID field has been empty because lsblk was being called before the kernel was ready. This commit fixed the issue (at least when I run it with `sudo ./mylfs.sh --init && sudo ./mylfs.sh --mount && cat mnt/lfs/boot/grub/grub.cfg`): https://github.com/krglaws/MyLFS/commit/c1efe708b4f91c55f5968816f51ff0b08daef456
